### PR TITLE
DN-3383

### PR DESCRIPTION
**Pull request by marciogualtieri-polyswarm**
DN-3383: Run micro-engine's blocking code inside an executor.

### CODE REVIEW

File: src/polyswarmclient/utils.py [hunk 1]
```diff
@@ -1,20 +1,22 @@
 import asyncio
+import base58
```
Comment (alanjds-ps) [inline src/polyswarmclient/utils.py]: Why was this moved up?

File: src/polyswarmclient/utils.py [hunk 2]
```diff
@@ -139,3 +147,16 @@ def is_valid_ipfs_uri(ipfs_uri):
...
+def run_in_executor(f):
+    """
+    Decorator. Runs the function it decorates in an executor.
+    """
+    def inner(*args, **kwargs):
+        if sys.platform == 'win32':
+            loop = asyncio.ProactorEventLoop()
+        else:
+            loop = asyncio.get_event_loop()
+        return loop.run_in_executor(ThreadPoolExecutor(), lambda: f(*args, **kwargs))
```
Comment (alanjds-ps) [inline src/polyswarmclient/utils.py]: Suggestion: If the loop can be got only once, can detect the platform outside `inner`.:
```py
if sys.platform == 'win32':
    loop = asyncio.ProactorEventLoop()
else:
    loop = asyncio.get_event_loop()

def inner(*args, **kwargs):
    return loop.run_in_executor(ThreadPoolExecutor(), lambda: f(*args, **kwargs))

return inner

File: src/polyswarmclient/utils.py [hunk 3]
```diff
@@ -139,3 +147,16 @@ def is_valid_ipfs_uri(ipfs_uri):
     except Exception as err:
         logger.exception('Unexpected error: %s', err)
     return False
+
+
+def run_in_executor(f):
+    """
+    Decorator. Runs the function it decorates in an executor.
+    """
+    def inner(*args, **kwargs):
```
Comment (alanjds-ps) [inline src/polyswarmclient/utils.py]: Is a [good pratice](https://docs.python.org/3.8/library/functools.html#functools.wraps) to not lose [the metadata](https://stackoverflow.com/a/55102697/798575) of the decorated func when decorating it:
```suggestion
    @functools.wrap(f)
    def inner(*args, **kwargs):
```
(if accepting the suggestion, please remember to `import functools` ;) )

**Comment by marciogualtieri-polyswarm** [comment]
Realized that I don't need this change anymore, but thank you for the review, @alanjds-ps.
